clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java

```diff
@@ -158,9 +165,16 @@ public InetAddress currentAddress(String id) throws UnknownHostException {
      */
     public void disconnected(String id, long now) {
         NodeConnectionState nodeState = nodeState(id);
-        nodeState.state = ConnectionState.DISCONNECTED;
         nodeState.lastConnectAttemptMs = now;
-        updateReconnectBackoff(nodeState);
+        nodeState.failedAttempts++;
+        nodeState.reconnectBackoffMs = reconnectBackoff.term(nodeState.failedAttempts - 1);
+        if (nodeState.state == ConnectionState.CONNECTING) {
+            nodeState.connectionSetupTimeoutMs = connectionSetupTimeout.term(nodeState.failedAttempts);
```

Comment:
`failedAttempts` isn't really the number of failed connections, so not sure what we are setting timeout to.

Reply:
Correct. I'll add a new class property to record connection failures separately.